sympy/functions/special/delta_functions.py

```diff
@@ -218,12 +234,25 @@ def eval(cls, arg):
             return S.Zero
         elif arg.is_positive:
             return S.One
+        elif arg.is_zero and H0 != S.NaN:
+            return H0
 
-    def _eval_rewrite_as_Piecewise(self, arg):
+    def _eval_rewrite_as_Piecewise(self, *args):
+        if len(args) == 2:
+            arg, H0 = args[0], args[1]
+        else:
+            arg, H0 = args[0], None
         if arg.is_real:
-            return Piecewise((1, arg > 0), (S(1)/2, Eq(arg, 0)), (0, True))
+            if H0 == None:
+                return Piecewise((1, arg > 0), (0, arg < 0))
+            return Piecewise((1, arg > 0), (H0, Eq(arg, 0)), (0, True))
 
-    def _eval_rewrite_as_sign(self, arg):
+    def _eval_rewrite_as_sign(self, *args):
+        if len(args) == 2:
+            arg, H0 = args[0], args[1]
+        else:
+            arg, H0 = args[0], None
+        # the value of H0 is ignored currently
```

Comment:
Just a thought, but perhaps this method should *fail* unless `H0 == S.Half`?  But that could be a new PR...